src/day1.rs

```diff
@@ -65,14 +60,39 @@ fn part2(input: &str) -> String {
         let first_number = first.parse::<u32>().unwrap();
         let second_number = second.parse::<u32>().unwrap();
 
-        *first_list.entry(first_number).or_insert(0) += 1;
-        *second_list.entry(second_number).or_insert(0) += 1;
+        //    *first_list.entry(first_number).or_insert(0) += 1;
+        //    *second_list.entry(second_number).or_insert(0) += 1;
+
+        // A common pattern in rust is to use a match statement to switch based on if a value is present or not.
+        // BTW: Your old way is more efficient and almost certainly better especially if the value is not a simple integer.
+        // It looks like references are common when dealing with map entries in rust, I just haven't needed to use them too much...
+        let current_first = match first_list.get(&first_number) {
+            Some(n) => n.to_owned(),
```

Comment:
Need the `to_owned()` here so that the return type from the match statement is the same as the `0`.